src/altinn-app-frontend/src/features/form/components/Label.tsx

```diff
@@ -34,12 +31,17 @@ export default function Label(props: IFormLabelProps) {
           data-testid={`label-${props.id}`}
         >
           {props.labelText}
-          {shouldShowRequiredMarking &&
-            <RequiredIndicator />
-          }
-          {shouldShowOptionalMarking &&
-            <OptionalIndicator />
-          }
+          <RequiredIndicator
+            required={props.required}
+            readOnly={props.readOnly}
+            language={props.language}
+          />
+          <OptionalIndicator
+            labelSettings={props.labelSettings}
+            language={props.language}
+            readOnly={props.readOnly}
+            required={props.required}
+          />
```

Comment:
Is there a need for this change? I find that having the checks here in Label.tsx gives a much more verbose and clear picture of when the different indicators are shown. We also let the indicator components fetch from the store directly to avoid unnecessary prop drilling. But I'm open to making this change if there are good reasons for it 😄

Reply:
The reason I changed it was to avoid having to copy the logic for when to show hide the Optional indicator every time you want to use it. This logikk is written in 3 different components (AdressLabel, Label and Legend) and in my opinion it should be onyl written one place :P Another option could be to have a helper function for when to hid/show the OptionalIndicator outside the component.

I can of course remove it if you don't see the need for it :) You decide!

Reply:
Ok, I see your point :) Let's leave it as you have implemented it.